src/plugins/lua/known_senders.lua

```diff
@@ -72,6 +82,49 @@ local function make_key(input)
   return hash:hex()
 end
 
+local function make_key_replies(goop, sz, prefix)
+  local h = rspamd_cryptobox_hash.create()
+  h:update(goop)
+  local key = (prefix or '') .. h:base32():sub(1, sz)
+  return key
+end
+
+local zscore_script_id
+
+local function configure_scripts(_, _, _)
+  -- script checks if given recipients are in the local replies set of the sender
+  local redis_zscore_script = [[
+    local results = {}
+    local replies_recipients_addrs = {}
+    replies_recipients_addrs = ARGV
+    if replies_recipients_addrs ~= nil then
+
+      for _, rcpt in ipairs(replies_recipients_addrs) do
+        local score = redis.call('ZSCORE', KEYS[1], rcpt)
```

Comment:
This may not work with previous versions of redis and does not work on ubuntu. So I decided to keep this in the for loop instead of using the 'ZMSCORE' command.